### PR TITLE
Meta milestone pt2

### LINKED ISSUE #38: Decide on basic code style guidelines
Since this is multi-ppl project, a decision has to be made how code should be formatted on repo. Feel free to post suggestions below - I'll edit first post as discussion progresses.

- [x] Decide on braces
- [x] Decide on tabs vs spaces
- [x] Decide on folding
- [x] Decide on prefixes
- [x] Decide on include guards
- [x] Decide on IN/OUT
- [x] Apply on all source files
- [ ] Create wiki page and/or doc file

# Brace omitting - not allowed
Pros:
- shorter code
- improved readability for some ppl (me)

Cons:

- expanding statement to block after control expression makes line with condition being highlighted in diff, even if condition hasn't changed - **this is a major reason to write braces at all times**.
- @approxit hates it for some unknown reason

# Tabs vs spaces

- @approxit likes tab sized to 4 spaces, I like 2 spaces, so we're having source tab-based and its size set to different size in one's IDE.

# Code folding

Code is horizontally limited to 80 chars, so naturally some code folding must be done.
For functions in C files:

``` c
void fnShort(t1 arg1, t2 fnNameAndArgsFitInOneLine);

void fnWithManyArgs(
  t1 arg1, t2 arg2, t3 allArgsFitInOneLineButWithoutFnName
);

void fnWithTooManyArgs(
  t1 arg1, t2 arg2, t3 thereAreTooManyFnArgsToFitInOneLine,
  t4 arg4, t5 arg5, t6 soArgListIsBrokenToMultipleLines
);
```

Whereas in header files:

``` c
void fnVoid(void);

void fnNonVoid(
  t1 allArgsAre,
  t2 inSeparateLine
);
```

Other arg formatting stuff doesn't mix with variable tab size properly. So I guess it must stay this way.

# Naming conventions

camelCase with prefixes. List of prefixes:

- `ub` - unsigned byte (`UBYTE`, `uint8_t`)
- `uw` - unsigned word (`UWORD`, `uint16_t`)
- `ul` - unsigned long (`ULONG`, `uint32_t`)
-`b`, `w`, `l` for signed variants
- `p` - any pointer or array
- `cb` - any function pointer (callback)
- `t` - typedef
- `s` - struct instance
- `u` - union instance
- `f` - `float` or fixed-point var
- undecided prefixes for `double`, `uint64_t`

for global var scoping additionally:

- `g_` - global visible from other files
- `s_` - global visible from current file (`static`)

# Misc stuff

- all named structs and unions should be typedefed using following convention:

```c
typedef struct _tTypeName {

} tTypeName;
```

- ~~in header files, args should be documented using `IN`, `OUT` and `INOUT` defines:~~

``` c
void fn(
  UBYTE ubSomeInput,
  UBYTE *pSomeMoreInputDataLikeArrayOrSomething,
  UBYTE *pContentsGetOverwrittenOrArrayToBeFilled,
  UBYTE *pContentsAreUsedInFnCalculationsAndThenModified
);
```

- all fns in header files **must** be documented using DoxyComments. Functions which are not exported in headers _should_ be documented in .c file

- functions should be as short as possible. If needed, split function to multiple ones. There is no hard limit for fn length, but if it doesn't fit entirely on your screen then it's a good sign you're doing something wrong.

# Include guards

Currently we use something like that:

``` c
#ifndef GUARD_ACE_MANAGER_BLIT_H
#define GUARD_ACE_MANAGER_BLIT_H
```

Candidate for replacement:

``` c
#ifndef _ACE_MANAGERS_BLIT_H_
#define _ACE_MANAGERS_BLIT_H_
```

Pros:

- not flooding intellisense starting with "G"
- reflects filesystem location: source file in `inc/ace/managers/blit.h`

### LINKED ISSUE #54: Wiki vs "doc" folder
We should determine if we use wiki or docs folder. Here's [some arguments](https://www.dzombak.com/blog/2015/03/Why-use-GitHub-Wikis-.html) against wiki.

If we're dropping wiki, [relative links](https://blog.github.com/2013-01-31-relative-links-in-markup-files/) will come in handy.

- [x] Decide whether we use wiki or `doc` folder
- [ ] Port all wiki content to `doc` folder, if wiki is going to be scrapped

### PR DESCRIPTION
**Pull request by tehKaiN**
Fixes #38 
Fixes #54

### CODE REVIEW

File: include/ace/managers/mouse.h [hunk 1]
```diff
@@ -99,7 +93,7 @@ static inline void mouseSetBounds(
  * @return Mouse's current X position relative to top-left screen pos.
  */
 static inline UWORD mouseGetX(
-	IN UBYTE ubMousePort
+	UBYTE ubMousePort
```
Comment (approxit) [inline include/ace/managers/mouse.h]: It can be in line 95

File: include/ace/managers/mouse.h [hunk 2]
```diff
@@ -110,7 +104,7 @@ static inline UWORD mouseGetX(
  * @return Mouse's current X position relative to top-left screen pos.
  */
 static inline UWORD mouseGetY(
-	IN UBYTE ubMousePort
+	UBYTE ubMousePort
```
Comment (approxit) [inline include/ace/managers/mouse.h]: It can be in line 106

File: include/ace/managers/mouse.h [hunk 3]
```diff
@@ -192,9 +175,7 @@ static inline UBYTE mouseInRect(
  * @see mouseMoveBy()
  */
 static inline void mouseSetPosition(
-	IN UBYTE ubMousePort,
-	IN UWORD uwNewX,
-	IN UWORD uwNewY
+	UBYTE ubMousePort, UWORD uwNewX, UWORD uwNewY
```
Comment (approxit) [inline include/ace/managers/mouse.h]: Can be in line 177

File: include/ace/utils/chunky.h [hunk 4]
```diff
@@ -1,16 +1,16 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ * file, You can obone at http://mozilla.org/MPL/2.0/. */
```
Comment (approxit) [inline include/ace/utils/chunky.h]: Replace-all typo?

File: include/ace/utils/chunky.h [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
+#ifndef _ACE_UTILS_CHUNKY_H_
+#define _ACE_UTILS_CHUNKY_H_
 
 #include <ace/types.h>
 #include <ace/utils/bitmap.h>
 #include <fixmath/fix16.h>
 
 /**
- * @brief Returns color indices for 16 colors in a row starting from supplied
+ * @brief Returns color indices for 16 colora row starting from supplied
```
Comment (approxit) [inline include/ace/utils/chunky.h]: Replace-all typo?

File: include/ace/utils/chunky.h [hunk 6]
```diff
@@ -55,20 +46,16 @@ UBYTE chunkyFromPlanar(
  *
  * @param pSource   Source chunky pixels.
  * @param pDest     Destination chunky pixels.
- * @param fSin      Sine value of rotation angle.
+ * @param      Sine value of rotation angle.
```
Comment (approxit) [inline include/ace/utils/chunky.h]: Replace-all typo?

File: include/ace/utils/chunky.h [hunk 7]
```diff
@@ -77,20 +64,15 @@ void chunkyRotate(
  * This function assumes that chunky pixels are of same depth as bitmap.
  * Higher chunky bits will thus be ignored.
  *
- * @param pIn  Source chunky pixels.
+ * @param Source chunky pixels.
```
Comment (approxit) [inline include/ace/utils/chunky.h]: Replace-all typo?

File: include/ace/utils/font.h [hunk 8]
```diff
@@ -101,14 +94,11 @@ tTextBitMap *fontCreateTextBitMap(
...
-	IN const tFont *pFont,
-	IN const char *szText
+	const tFont *pFont, const char *szText
 );
 
 void fontFillTextBitMap(
-	IN const tFont *pFont,
-	INOUT tTextBitMap *pTextBitMap,
-	IN const char *szText
+	const tFont *pFont, tTextBitMap *pTextBitMap,	const char *szText
```
Comment (approxit) [inline include/ace/utils/font.h]: Tab before `const char`?

File: showcase/src/menu/menulist.h [hunk 9]
```diff
@@ -42,56 +42,36 @@ typedef struct _tMenuList {
 } tMenuList;
 
 typedef void (*tMenuSelectCb)(
-	IN struct _tMenuList *pList,
-	IN UBYTE ubPosIdx
+	struct _tMenuList *pList, UBYTE ubPosIdx
```
Comment (approxit) [inline showcase/src/menu/menulist.h]: Can be in line 44

File: include/ace/utils/chunky.h (hunk 4 above)
Comment (tehKaiN) [reply include/ace/utils/chunky.h]: nice!

File: include/ace/managers/mouse.h (hunk 3 above)
Comment (tehKaiN) [reply include/ace/managers/mouse.h]: Doesn't fit in 80col limit

File: showcase/src/menu/menulist.h (hunk 9 above)
Comment (tehKaiN) [reply showcase/src/menu/menulist.h]: fixed in 5c6b2dd

File: include/ace/utils/font.h (hunk 8 above)
Comment (tehKaiN) [reply include/ace/utils/font.h]: fixed in 5c6b2dd

File: include/ace/utils/chunky.h (hunk 7 above)
Comment (tehKaiN) [reply include/ace/utils/chunky.h]: fixed in 5c6b2dd

File: include/ace/utils/chunky.h (hunk 6 above)
Comment (tehKaiN) [reply include/ace/utils/chunky.h]: fixed in 5c6b2dd

File: include/ace/utils/chunky.h (hunk 5 above)
Comment (tehKaiN) [reply include/ace/utils/chunky.h]: fixed in 5c6b2dd

File: include/ace/managers/mouse.h (hunk 2 above)
Comment (tehKaiN) [reply include/ace/managers/mouse.h]: fixed in 5c6b2dd

File: include/ace/managers/mouse.h (hunk 1 above)
Comment (tehKaiN) [reply include/ace/managers/mouse.h]: fixed in 5c6b2dd

File: include/ace/managers/copper.h [hunk 10]
```diff
@@ -197,10 +180,7 @@ void copProcessBlocks(void);
  *
  *  @return Pointer to resulting copBlock.
  */
-tCopBlock *copBlockDisableSprites(
-	IN tCopList *pList,
-	IN FUBYTE fubSpriteMask
-);
+tCopBlock *copBlockDisableSprites(tCopList *pList,FUBYTE fubSpriteMask);
```
Comment (approxit) [inline include/ace/managers/copper.h]: Space! :D

File: include/ace/managers/viewport/simplebuffer.h [hunk 11]
```diff
@@ -73,26 +70,16 @@ tSimpleBufferManager *simpleBufferCreate(
  *
  *  @todo Realloc copper buffer to reflect BPP change.
  */
-void simpleBufferSetBitmap(
-	IN tSimpleBufferManager *pManager,
-	IN tBitMap *pBitMap
-);
+void simpleBufferSetBitmap(tSimpleBufferManager *pManager,	tBitMap *pBitMap);
```
Comment (approxit) [inline include/ace/managers/viewport/simplebuffer.h]: Tab before `tBitMap`

File: include/ace/managers/copper.h (hunk 10 above)
Comment (tehKaiN) [reply include/ace/managers/copper.h]: fixed in d5c23eb

File: include/ace/managers/viewport/simplebuffer.h (hunk 11 above)
Comment (tehKaiN) [reply include/ace/managers/viewport/simplebuffer.h]: fixed in 186f3aa

**Pull request merged**